nodejs/src/stream_manager.rs

```diff
@@ -0,0 +1,575 @@
+// Copyright Pravega Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use neon::prelude::*;
+use pravega_client::client_factory::ClientFactory;
+use pravega_client_config::{ClientConfig, ClientConfigBuilder};
+use pravega_client_retry::retry_result::RetryError;
+use pravega_client_shared::*;
+use pravega_controller_client::ControllerError;
+use tracing::info;
+
+///
+/// An internal rust struct that holds the necessary info to perform actions on StreamManager.
+/// The `js_new` method will return a boxed(wrapped) StreamManager as an external object.
+///
+pub(crate) struct StreamManager {
+    controller_ip: String,
+    cf: ClientFactory,
+    config: ClientConfig,
+}
+
+pub(crate) struct StreamRetentionPolicy {
+    retention: Retention,
+}
+
+impl Finalize for StreamRetentionPolicy {}
```

Comment:
Is this needed?

Reply:
`Finalize` trait is a must. See here https://docs.rs/neon/0.10.0/neon/types/trait.Finalize.html

It works like a `Drop` trait and will release other manual-controlled objects if needed.

PS: I do see a destructor in StreamWriter of the Python binding, but I can't find a way to do the same thing in neon. Will figure it out later.